ledger/src/blockstore_db.rs

```diff
@@ -520,7 +518,7 @@ impl Rocks {
     }
 
     fn is_primary_access(&self) -> bool {
-        self.1 == ActualAccessType::Primary
+        self.1 != AccessType::Secondary
```

Comment:
Probably safer with `self.1 == AccessType::Primary || self.1 == AccessType::PrimaryForMaintenance`.  If we introduce a new AccessType in the future, then this function might be something easily forgotten.

Reply:
100% agree